lxd/daemon.go

```diff
@@ -1350,8 +1347,8 @@ func (d *Daemon) NodeRefreshTask(heartbeatData *cluster.APIHeartbeat) {
 	for i, node := range heartbeatData.Members {
 		// Exclude nodes that haven't been heard from for a while.
 		// This is to avoid forkdns delaying results by querying an offline node.
-		if time.Since(node.LastHeartbeat) > (time.Second * 30) {
-			logger.Warnf("Removing out of date node: %+v", node)
+		if time.Since(node.LastHeartbeat) > (time.Second * cluster.HeartbeatInterval * 3) {
```

Comment:
Just in case github ate my previous comment as this commit updates the same chunk of code.
I think we should use `cluster.offline_threshold` for that which is user configurable.

Reply:
Fixed. I have added the concept of a "partial node list" for heartbeats, and a concept for the leader of "last heartbeat send time", as I found that during a leader election, this can take 20s and means the first heartbeat sent after that will sent out of date node lists to the nodes (resulting in all forkdns peers being removed until the next heartbeat).

Instead the leader detects that they havent sent a heartbeat before/or > offline threshold, and sends only the raft members with an indicator this is a partial heartbeat list.

I then use this same indicator for node joining rather than `SrcLeader`.